src/container/App.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import './App.css';
+import Display from '../components/Display/Display';
+import ButtonPanel from '../components/ButtonPanel/ButtonPanel';
+import calculate from '../logic/calculate';
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      calculation: '0',
+      total: '',
+      current: '',
+      operator: ''
+    }
+  }
+
+
+  render() {
+
+    const handleClick = (btnName) => {
+      const data = {
+        calculation: this.state.calculation,
+        total: this.state.total,
+        current: this.state.current,
+        operator: this.state.operator
+      }
+      const newData = calculate(data, btnName);
```

Comment:
To make this cleaner we can use object destructuring for the newData:

```
const { calculation, total, current, operator } = newData;

this.setState = ({ calculation, total, current, operator });
```

Learn more about this [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). Further, had we used a linter we could have easily converted everything into a better format :)